AxioVision Converter

joeykleingers opened the pull request:
* Adding a filter that can convert AxioVision files to either Json or text files.

joeykleingers committed:
Adding AxioVision to Json filter and writing the filter contents.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Writing unit test for AxioVisionV4ToJson filter.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Writing documentation for AxioVisionV4ToJson filter.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Cleaning up unneeded includes and unused variables.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Rename AxioVisionV4ToJson to ITKAxioVisionV4Converter.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Pulling conversion algorithm out into its own helper class.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Adding option to convert AxioVision file to a text file.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

imikejackson reviewed (changes requested):
* Did we talk about maybe converting the Axio Vision XML into more human readable XML?
* We should also have some sort of delimiter in the text based output so we can tell the difference between the key and the value. Semicolon or maybe the pipe (|) character? Did you use tabs?
* I think each section in the text file should start and end with some kind of marker: # START [SECTION NAME] and # END [SECTION NAME]. This will make it easier to know when to stop parsing a section.

joeykleingers committed:
Updating text file conversion algorithm to implement requested features.

Added an ending marker to each section and updated the marker text.
Changed the delimiter from tabs to |.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Updating marker and group text for output text and json files.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Adding option to convert to a more human-readable XML file.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Updating documentation.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>
